lhotse/recipes/commonvoice.py

```diff
@@ -114,6 +112,112 @@ def download_commonvoice(
         completed_detector.touch()
 
 
+def _read_cv_manifests_if_cached(
+    output_dir: Optional[Pathlike],
+    language: str,
+) -> Dict[str, Dict[str, Union[RecordingSet, SupervisionSet]]]:
+    """
+    Returns:
+        {'train': {'recordings': ..., 'supervisions': ...}, 'dev': ..., 'test': ...}
+    """
+    if output_dir is None:
+        return {}
+    manifests = defaultdict(dict)
+    for part in ["train", "dev", "test"]:
+        for manifest in ["recordings", "supervisions"]:
+            path = output_dir / f"cv_{manifest}_{language}_{part}.jsonl.gz"
+            if not path.is_file():
+                continue
+            manifests[part][manifest] = load_manifest(path)
+    return manifests
+
+
+def _parse_utterance(
+    lang_path: Path,
+    language: str,
+    audio_info: list,
+) -> Optional[Tuple[Recording, SupervisionSegment]]:
+    audio_path = lang_path / "clips" / audio_info[1]
+
+    if not audio_path.is_file():
+        logging.info(f"No such file: {audio_path}")
+        return None
+
+    recording_id = Path(audio_info[1]).stem
+    recording = Recording.from_file(path=audio_path, recording_id=recording_id)
+
+    segment = SupervisionSegment(
+        id=recording_id,
+        recording_id=recording_id,
+        start=0.0,
+        duration=recording.duration,
+        channel=0,
+        language=language,
+        speaker=audio_info[0],
+        text=audio_info[2].strip(),
+        gender=audio_info[6],
+        custom={
+            "age": audio_info[5],
+            "accents": audio_info[7],
+        },
+    )
+    return recording, segment
+
+
+def _prepare_part(
+    lang: str,
+    part: str,
+    lang_path: Pathlike,
+    num_jobs: int = 1,
+) -> Tuple[RecordingSet, SupervisionSet]:
+    """
+    Prepares part of CommonVoice data.
+
+    :param lang: string language code (e.g., "en").
+    :param part: which split to prepare (e.g., "train", "validated", etc.).
+    :param lang_path: path to a CommonVoice directory for a specific language
+        (e.g., "/path/to/cv-corpus-13.0-2023-03-09/pl").
+    :param num_jobs: How many concurrent workers to use for scanning of the audio files.
+    :return: a tuple of (RecordingSet, SupervisionSet) objects opened in lazy mode,
```

Comment:
This doc is outdated with your changes because the manifests are no longer created or opened lazily here. That may make the recipe quite memory-intensive; would it be possible to keep the `.open_writer()` mechanism with your changes? You can open the writers like `RecordingSet.open_writer()` around line 206-207 and instead of appending to a list, write to disk.

Reply:
Sure